### PR TITLE
MGDOBR-1289: add create-processor-v2.sh script

### PR DESCRIPTION
**Pull request by kostola**
[MGDOBR-1289](https://issues.redhat.com/browse/MGDOBR-1289)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by robpblake — APPROVED** [review]
Tested on a mac. All good 👍 Thank you!

**Review by r00ta — COMMENTED** [review]
`Error: unknown command <THE FILENAME> for "yq"`

**Review by r00ta — COMMENTED** [review]
looks like for 3.4.0 the command is `yq read <filename> -j`. Do you have `yq read` in 4.x?

File: dev/bin/interaction/create-processor-v2.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  echo "$0: destination bridge ID is empty"; usage_and_exit 1;
+elif [ ! "${processor_name}" ]; then
+  echo "$0: processor name ID is empty"; usage_and_exit 1;
+elif [ ! "${flow_file}" ]; then
+  echo "$0: flow file is empty"; usage_and_exit 1;
+elif [ ! -f "${flow_file}" ]; then
+  echo "$0: flow file \"${flow_file}\" doesn't exist"; usage_and_exit 1;
+fi
+
+json_flow=$( yq -o=json "$flow_file" )
```
Comment (danielezonca) [inline dev/bin/interaction/create-processor-v2.sh:55]: If the file is malformed the script should stop, I tried with a not valid yaml and I got

```
 Error: bad file 'README.MD': yaml: line 20: did not find expected <document start>
 + curl -vvv http://localhost:8080/api/smartevents_mgmt/v2/bridges/test/processors -H 'Authorization: bearer ' -H 'Accept: application/json' -H 'Content-Type: application/json' --data-raw '{"name":"test","flows":"Start by making a copy of the `environment-example` file called `environment` and modify it accordingly, it won'\''t be pushed into the repository.\nThen it'\''s important to always have the bearer token set\n```shell\nsource get-token.sh\n```\nAnd then call the others such as\n```shell\n./get-all-bridges.sh ```"'
 *   Trying 127.0.0.1:8080...
 * connect to 127.0.0.1 port 8080 failed: Connection refused
 *   Trying ::1:8080...
 * connect to ::1 port 8080 failed: Connection refused
 * Failed to connect to localhost port 8080 after 10 ms: Connection refused
 * Closing connection 0
 curl: (7) Failed to connect to localhost port 8080 after 10 ms: Connection refused
```

Comment (kostola) [reply dev/bin/interaction/create-processor-v2.sh]: added a check for parse failures that blocks the script before executing the API call

**Comment by kostola** [comment]
> Looks like there are many differences between 3.x and 4.x https://mikefarah.gitbook.io/yq/upgrading-from-v3
> 
> You might want to add 4.x as required dependency or check the version and run different commands in the scri accordingly. No strong opinions by me

@r00ta I added a `YQ_VERSION` env variable that allows to explicitly force the `yq` v3 syntax. Try calling it with:

```
YQ_VERSION=3 ./create-processor-v2.sh ...
```

**Review by r00ta — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]

**Pull request merged**